Added a score based on the distance travelled

marc-jones opened the pull request.

marc-jones committed:
Added a score based on the distance travelled

tomcraven commented on src/states/game.js:
```diff
@@ -24,9 +31,20 @@ Game.prototype =
 
+    preload : function()
+    {
+        this.load.script('webfont',
```
This would mean the game couldn't be played offline (or at least - the score couldn't be displayed offline), is there a way of bundling this with the game files? Is that something we're bothered about? I'm thinking if we want to create a phone app for this that doesn't require a data connection. Maybe this is something that we don't need to address for the prototype.

If we do go down this route, we should register loading this with the preload state, like we do for the images. This gives us quick loading times between states as it's all done during start up. It also means we're not trying to reload this script each time we go from gameover screen back to the main game.

tomcraven commented on src/states/game.js:
```diff
@@ -61,6 +80,17 @@ Game.prototype =
+        this.offset_x = 0;
     },
 
     update : function()
```
Getting a bit bloated in the update function, might be a good idea to split this out into different funcs

tomcraven commented on src/states/game.js:
```diff
@@ -24,9 +31,20 @@ Game.prototype =
         ship_mass: 1,
         blackHole_mass: 500000,
-        camera_cutoff_x: 500
+        camera_cutoff_x: 500,
```
I like these params, nice to be able to fiddle with them in one place!

tomcraven commented:
I like the new starting positions for the player and black hole. It has the nice effect of propelling the player forwards if you don't press anything!

marc-jones committed:
Removed the dependence on an Internet connection for font loading

To do this required converting the .ttf font file to a .png and
.fnt file using an online tool available at kvazars.com/littera/

tomcraven commented on src/states/game.js:
```diff
@@ -24,9 +24,21 @@ Game.prototype =
 
+    preload : function()
+    {
+        this.load.bitmapFont('RobotoMono-Regular',
```
preload is called when we switch to the 'game' state. It'll be called every time we switch. We don't have to load the font each time we switch to the game state, we only have to load it once (though you would hope that phaser caches which ones it's loaded and which one's it hasn't, but you never know). We should do this load in the 'preload' state's preload function. Similar to how images work (registering them with the preload state in onload).

It would also be good to pull `RobotoMono-Regular` into a variable so we can reuse it when we create the text further down:

```
this.scoreText = bitmapText( centerX, 45, this.font_name, ... )
```

marc-jones committed:
Moved the font loading to the preload state

marc-jones commented:
@tomcraven Is there somewhere I could put the variable storing the font name string so that both preload.js and game.js can see it? Or do you think it's better to have the font name variable specified in game.js where the other parameters are specified like I have it?

tomcraven commented:
Happy to see font name (and file paths?) in the assets structure with the images. I suppose fonts are assets too

marc-jones commented:
Ready to merge then?

tomcraven commented:
yea looks good!
